src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -465,44 +473,34 @@ private BoundBlock MakeExpressionUsingStatement(BoundUsingStatement node, BoundB
         /// <summary>
         /// Synthesize a call `expression.Method()`, but with some extra smarts to handle extension methods, and to fill-in optional and params parameters.
         /// </summary>
-        private BoundExpression MakeCallWithNoExplicitArgument(Binder binder, SyntaxNode syntax, BoundExpression expression, MethodSymbol method)
+        private BoundExpression MakeCallWithNoExplicitArgument(MethodArgumentInfo methodArgumentInfo, SyntaxNode syntax, BoundExpression? expression)
         {
-            var receiver = method.IsExtensionMethod ? null : expression;
+            MethodSymbol method = methodArgumentInfo.Method;
 
-            var argsBuilder = ArrayBuilder<BoundExpression>.GetInstance();
+#if DEBUG
+            Debug.Assert(!method.IsExtensionMethod || method.ParameterRefKinds.IsDefaultOrEmpty || method.ParameterRefKinds[0] != RefKind.Ref);
```

Comment:
>method.ParameterRefKinds.IsDefaultOrEmpty [](start = 54, length = 41)

Is this method able to handle ref/out/in parameters?

Reply:
If I added them to `MethodArgumentInfo`, then it probably could. However, that isn't necessary anywhere we want to use this today.

---
In reply to: [549857233](https://github.com/dotnet/roslyn/pull/50105#discussion_r549857233) [](ancestors = 549857233)

Reply:
I also am not sure what to make of a call that had no explicit arguments and also had by-ref parameters.

---
In reply to: [549861482](https://github.com/dotnet/roslyn/pull/50105#discussion_r549861482) [](ancestors = 549861482,549857233)

Reply:
> If I added them to MethodArgumentInfo, then it probably could. However, that isn't necessary anywhere we want to use this today.

Perhaps we should assert unconditionally that there are no such parameters.

---
In reply to: [549861770](https://github.com/dotnet/roslyn/pull/50105#discussion_r549861770) [](ancestors = 549861770,549861482,549857233)

Reply:
> Perhaps we should assert unconditionally that there are no such parameters.

Actually, I was incorrect: we do support one refkind, `in` on the first parameter. However, passing default as the refkinds parameter is the correct thing to do since we did not have an `in` in source, and it will be handled by `GetEffectiveArgumentRefKinds` during the process of `MakeCall`.

---
In reply to: [549870955](https://github.com/dotnet/roslyn/pull/50105#discussion_r549870955) [](ancestors = 549870955,549861770,549861482,549857233)

Reply:
I can tighten the assertion, however.

---
In reply to: [549874598](https://github.com/dotnet/roslyn/pull/50105#discussion_r549874598) [](ancestors = 549874598,549870955,549861770,549861482,549857233)